flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/AvroSchemaConvertor.java

```diff
@@ -0,0 +1,168 @@
+                if (logicalDataType != null) {
+                    return logicalDataType;
+                }
+                return DataTypes.BIGINT().notNull();
+            case FLOAT:
+                return DataTypes.FLOAT().notNull();
+            case DOUBLE:
+                return DataTypes.DOUBLE().notNull();
+            case BOOLEAN:
+                return DataTypes.BOOLEAN().notNull();
+            case NULL:
+                return DataTypes.NULL();
+        }
+        throw new IllegalArgumentException("Unsupported Avro type '" + schema.getType() + "'.");
+    }
+
+    /**
+     * Function to convert Avro Schema Field value to Data Type (Tale API Schema).
+     *
+     * @param fieldSchema Avro Schema describing the schema for the value.
+     * @return Converted {@link DataType} value if supported logical type exists, NULL Otherwise.
+     */
+    private static DataType handleLogicalTypeSchema(Schema fieldSchema) {
+        String logicalTypeString = fieldSchema.getProp(LogicalType.LOGICAL_TYPE_PROP);
+        if (logicalTypeString != null) {
+            // 1. In case, the Schema has a Logical Type.
+            if (logicalTypeString.equals(LogicalTypes.date().getName())) {
+                return DataTypes.DATE().notNull();
+            } else if (logicalTypeString.equals(LogicalTypes.decimal(1).getName())) {
+                final LogicalTypes.Decimal decimalType =
+                        (LogicalTypes.Decimal) fieldSchema.getLogicalType();
+                int precision = decimalType.getPrecision();
+                return DataTypes.DECIMAL(precision, decimalType.getScale()).notNull();
+            } else if (logicalTypeString.equals(LogicalTypes.timestampMicros().getName())) {
+                return DataTypes.TIMESTAMP(6).notNull();
+            } else if (logicalTypeString.equals(LogicalTypes.timestampMillis().getName())) {
+                return DataTypes.TIMESTAMP(3).notNull();
+            } else if (logicalTypeString.equals(LogicalTypes.uuid().getName())) {
+                return DataTypes.STRING().notNull();
+            } else if (logicalTypeString.equals(LogicalTypes.timeMillis().getName())) {
+                return DataTypes.TIME(3).notNull();
+            } else if (logicalTypeString.equals((LogicalTypes.timeMicros().getName()))) {
+                return DataTypes.TIME(6).notNull();
+            } else if (logicalTypeString.equals(LogicalTypes.localTimestampMillis().getName())) {
+                return DataTypes.TIMESTAMP_WITH_LOCAL_TIME_ZONE(3).notNull();
+            } else if (logicalTypeString.equals(LogicalTypes.localTimestampMicros().getName())) {
+                /// TIMESTAMP_LTZ() => has precision 6 by default.
+                return DataTypes.TIMESTAMP_WITH_LOCAL_TIME_ZONE().notNull();
+            } else if (logicalTypeString.equals("geography_wkt")
+                    || logicalTypeString.equals("Json")) {
+                return DataTypes.STRING().notNull();
+            }
```

Comment:
i imagine the answer is "no" but can this block be a switch statement?

Reply:
Switch blocks cannot have variables in case statements only literals or constants.

Reply:
thanks! you can resolve this convo